### PR TITLE
Doc fixes

### PR DESCRIPTION
**Pull request by petdance**
Fixed example code which was using `my_ok` instead of `ok` for testing `my_ok` itself.

Cleaned up miscellaneous spelling errors, missing punctuation, run-on sentences and wording fixes throughout some docs.

### CODE REVIEW

File: lib/Test2/API.pm [hunk 1]
```diff
@@ -596,9 +596,9 @@ generated by the test system:
         my_ok(0, "fail");
     };
 
-    my_ok(@$events == 2, "got 2 events, the pass and the fail");
```
Comment (exodist) [inline lib/Test2/API.pm]: notice that ok() is never imported in this example. The example is intended to demonstrate that intercept { ... } lets you use your own test tools to test themselves.

**Comment by exodist** [comment]
Added an important note. No problem with the other changes.

I think the travis failures are my fault from a previous commit, I will have to fix that.

**Comment by exodist** [comment]
cherry-picked your second commit, but not the my_ok one.

File: lib/Test2/API.pm (hunk 1 above)
Comment (petdance) [inline lib/Test2/API.pm]: That makes sense.  I think you should add a note explaining explicitly, because it seems like a mistake.

**Pull request closed without merging**